requirements.txt

```diff
@@ -1,16 +1,22 @@
+black==23.1.0
 certifi==2022.12.7
-chardet==3.0.4
-idna==2.9
-numpy==1.23.4; python_version >= '3.8'
+chardet==5.0.0
+idna==3.4
+numpy==1.24.2; python_version >= '3.8'
 numpy==1.19.5; python_version < '3.8'
-mypy==0.790
-mypy-extensions==0.4.3
+mypy==1.0.1; python_version >= '3.8'
+mypy-extensions==1.0.0; python_version >= '3.8'
+mypy==0.790; python_version < '3.8'
+mypy-extensions==0.4.3; python_version < '3.8'
 pylint==2.8.3
```

Comment:
I think we do need to update pylint for 3.11, but will need to work through the warnings (rather than try and force wrapt to update)